File: web/classic/src/helpers/branding.js
export const BUILTIN_SYSTEM_NAME = 'RightMaaS';
export const LEGACY_SYSTEM_NAME = 'New API';
export const LEGACY_BRAND_NAME = 'RightToken';

export function resolveSystemName(value) {
  if (typeof value !== 'string') return BUILTIN_SYSTEM_NAME;
  const trimmed = value.trim();
  if (
    trimmed === '' ||
    trimmed === LEGACY_SYSTEM_NAME ||
    trimmed === LEGACY_BRAND_NAME
  ) {
    return BUILTIN_SYSTEM_NAME;
  }
  return trimmed;
}

export function replaceBrandNameTokens(value, systemName = BUILTIN_SYSTEM_NAME) {
  if (typeof value !== 'string') return value;
  const resolvedName = resolveSystemName(systemName);
  return value
    .split(BUILTIN_SYSTEM_NAME)
    .join(resolvedName)
    .split(LEGACY_BRAND_NAME)
    .join(resolvedName)
    .split(LEGACY_SYSTEM_NAME)
    .join(resolvedName);
}

File: common/branding.go
package common

import "strings"

const (
	BuiltInSystemName = "RightMaaS"
	LegacySystemName  = "New API"
	LegacyBrandName   = "RightToken"
)

func NormalizeSystemName(value string) string {
	trimmed := strings.TrimSpace(value)
	if trimmed == "" || trimmed == LegacySystemName || trimmed == LegacyBrandName {
		return BuiltInSystemName
	}
	return trimmed
}
